app_clients/cassandra-client/cassandra-liveness-check.sh

```diff
@@ -0,0 +1,198 @@
+{
+ if [[ $1 = "--help" || $1 = "-h" ]]; then
+     echo 
+     echo "Usage: bash $0"
+     echo
+     echo "Set these MANDATORY env before running the script"
+     echo 
+     echo "CASSANDRA_PORT              : Cassandra Transport Port"
+     echo "CASSANDRA_SVC_NAME          : Cassandra Service Name"
+     echo
+     echo "Set these Optional env before running the script"
+     echo
+     echo "LIVENESS_PERIOD_SECONDS : Liveness check interval (default: 10s)"
+     echo "LIVENESS_TIMEOUT_SECONDS: Liveness probe failure timeout (default: 10s)"
+     echo "LIVENESS_RETRY_COUNT    : Liveness probe failure retry count (default: 3)"
+     echo 
+     exit  
+ fi  
+}
+
+# Create the keyspace
+create_keyspace()
+{
+  echo "creating the keyspace"
+
+  for i in `seq 1 $L_R_C`; do
+   cqlsh $1 $2 -e "CREATE KEYSPACE IF NOT EXISTS $KEYSPACE_NAME WITH replication = {'class' : 'SimpleStrategy', 'replication_factor' : '3'};" > /dev/null 2>&1
+   rc=$?
+   [ $rc -eq 0 ] && break
+   sleep $L_T_S
+ done
+}
+
+# Create the table
+create_table()
+{
+  echo "creating the table"
+
+  for i in `seq 1 $L_R_C`; do
+   cqlsh $1 $2 -e "use $KEYSPACE_NAME; CREATE TABLE $TABLE_NAME( name text PRIMARY KEY, roll int );" > /dev/null 2>&1
+   rc=$?
+   [ $rc -eq 0 ] && break
+   sleep $L_T_S
+ done
+}
+
+# Insert data into table
+insert_data()
+{
+  echo "inserting data into table"
+
+  for i in `seq 1 $L_R_C`; do
+   cqlsh $1 $2 -e "use $KEYSPACE_NAME; INSERT INTO $TABLE_NAME (name, roll) VALUES('jamesbond',007);" > /dev/null 2>&1
+   rc=$?
+   [ $rc -eq 0 ] && break
+   sleep $L_T_S
+ done
+}
+
+# Print data from table
+print_data()
+{
+  echo "printing data from table"
+
+  for i in `seq 1 $L_R_C`; do
+   rc=0
+   cqlsh $1 $2 -e "use $KEYSPACE_NAME; SELECT * from $TABLE_NAME;" || rc=-1
+   
+   [ $rc -eq 0 ] && break
+   sleep $L_T_S
+ done
+}
+
+# Drop table
+drop_table()
+{
+  echo "Dropping table"
+
+  for i in `seq 1 $L_R_C`; do
+   cqlsh $1 $2 -e "use $KEYSPACE_NAME; DROP TABLE $TABLE_NAME;" > /dev/null 2>&1
+   rc=$?
+   [ $rc -eq 0 ] && break
+   sleep $L_T_S
+ done
+}
+
+# Drop keyspace 
+drop_keyspace()
+{
+  echo "Dropping keyspace"
+
+  for i in `seq 1 $L_R_C`; do
+   cqlsh $1 $2 -e "DROP KEYSPACE $KEYSPACE_NAME;" > /dev/null 2>&1
+   rc=$?
+   [ $rc -eq 0 ] && break
+   sleep $L_T_S
+ done
+}
+
+###########
+#  MAIN   #
+###########
+
+# Verify availability of Required Variables
+if [[ -z ${CASSANDRA_PORT} || -z $CASSANDRA_SVC_NAME} ]]; then
+ usage --help;
+fi
+
+# Begin Liveness check
+echo "starting Liveness check" 
+
+#Initialize the flag
+cycle=0
+
+while true
+do 
+ # setting flag to 0, marking beggining of cycle
+ #FLAG=0
+
+ # creating keyspace
+ create_keyspace ${Svc} ${Port}
+ if [ $rc -ne 0 ]; then
+   echo "Creation of keyspace failed after $(($L_R_C * $L_T_S))s, exiting" && break
+ fi
+
+ # creating table
+ create_table ${Svc} ${Port}
+ if [ $rc -ne 0 ]; then
+   echo "Creation of table failed after $(($L_R_C * $L_T_S))s, exiting" && break
```

Comment:
break will take an graceful exit into the Line 195 (print and end script). However, we want the script to end ungracefully when any of the CRUD ops are not successful. If you would like to break, then have the final step to exit with $rc like:  

exit ${rc}